src/auro-checkbox-group.js

```diff
@@ -223,37 +223,37 @@ export class AuroCheckboxGroup extends LitElement {
     this.items = Array.from(this.querySelectorAll(checkboxTagName));
 
     this.handlePreselectedItems();
-    this.handleCheckboxAttributes();
 
     this.validation.validate(this);
   }
 
-  /**
-   * Updates checkbox attributes for each item (private method).
-   * Sets the 'error' attribute if there is an error, and adjusts 'disabled' property.
-   * @private
-   * @returns {void}
-   */
-  handleCheckboxAttributes() {
-    this.items.forEach((el) => {
-      if (this.validity && this.validity !== 'valid') {
-        el.setAttribute('error', '');
-      } else {
-        el.removeAttribute('error');
-      }
-
-      el.disabled = this.disabled;
-    });
-  }
-
   /**
    * LitElement lifecycle method. Called after the DOM has been updated.
    * @param {Map<string, any>} changedProperties - Keys are the names of changed properties, values are the corresponding previous values.
    * @returns {void}
    */
   updated(changedProperties) {
-    if (changedProperties.has('disabled') || changedProperties.has('validity')) {
-      this.handleCheckboxAttributes();
+    if (changedProperties.has('disabled')) {
+      // Counts the number of checkboxes that have the disabled attribute
+      const checkboxDisabledCount = this.items.filter((el) => el.hasAttribute('disabled')).length;
+
+      this.items.forEach((el) => {
+        if (this.disabled) {
+          el.setAttribute('disabled', true);
+        } else if (checkboxDisabledCount === this.items.length) {
```

Comment:
Why is this check necessary? If `this.disabled` is changed and the change is that it was removed cant we just remove `disabled` from all checkboxes regardless of how many are currently disabled?

What is the user story where checkboxDisabledCount would not be equal in this case when the group is disabled?

Reply:
The reason this change is there is because if you have this function and just have code that removes the attribute from the individual checkboxes with a simple `else` statement when the group does not have the `disabled` attribute, then it the use of individual disabled will not work because the attribute will get removed upon load.

Reply:
Isn't that because you are setting `disabled = false` in the constructor? Why not just have `disabled = undefined` as the default?

Reply:
This check makes it so that the `disabled` attribute is only removed when ALL individual checkboxes have the `disabled` attribute

Reply:
Didnt even notice that, thanks!